### ISSUE TITLE
Failing to compile `build.gradle.kts` does not fail the Gradle build

### ISSUE THREAD

**Issue by big-guy** [gradle/gradle-script-kotlin#170]
Given a `build.gradle.kts`:

```
import org.gradle.api.*

task<DefaultTask>("broken)
```

Running `gradle -b build.gradle.kts help` produces:

> Error: Expecting '"' (/Users/sterling/gits/tmp/kotlin/build.gradle.kts (3:27))
> Error: Expecting ')' (/Users/sterling/gits/tmp/kotlin/build.gradle.kts (3:27))
> :help
> 
> Welcome to Gradle 3.3-20161028185232+0000.
> 
> To run a build, run gradle <task> ...
> 
> To see a list of available tasks, run gradle tasks
> 
> To see a list of command-line options, run gradle --help
> 
> To see more detail about a task, run gradle help --task <task>
> 
> BUILD SUCCESSFUL
> 
> Total time: 1.459 secs
> 

I would expect this build to fail.  If I look at `gradle tasks`, I see there's a task named `broken)`.

**Comment by big-guy** [comment]
And I should add, this isn't all types of failures.  This is just one case I found.  Other failures _do_ cause build failures.

**Comment by bamboo** [comment]
Curious :)

Thanks, @big-guy!

**Comment by mkobit** [comment]
A few other weird examples with `0.7.0`:

1. Example version as second argument in `kotlinModule`

```
plugins {
  `lifecycle-base`
  `build-dashboard`
  java
}

dependencies {
  compile(kotlinModule("stdlib", ))
}

tasks {
  "wrapper"(Wrapper::class) {
    distributionUrl = "https://repo.gradle.org/gradle/dist-snapshots/gradle-script-kotlin-3.4-20170117200919+0000-all.zip"
  }
}
```

Results in:

```
./gradlew build
:compileJava FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Could not resolve all dependencies for configuration ':compileClasspath'.
> Cannot resolve external dependency org.jetbrains.kotlin:kotlin-stdlib:1.1-M04 because no repositories are defined.
  Required by:
      project :

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.

BUILD FAILED

Total time: 0.755 secs
```

----

2. More contrived examples with a bunch of commas

```
plugins {
  `lifecycle-base`,,,,,,,,
  `build-dashboard`,,,,,
  java,,,
}

repositories {
  jcenter(),,,,
}
//

dependencies {
  compile(kotlinModule("stdlib", ))
}

tasks {
  "wrapper"(Wrapper::class) {
    distributionUrl = "https://repo.gradle.org/gradle/dist-snapshots/gradle-script-kotlin-3.4-20170117200919+0000-all.zip"
  }
}
```

**Comment by bamboo** [comment]
Resolved in 56828939fd16191516c29b574bfc6bf721f5f424.
